src/test/proto/test.proto

```diff
@@ -46,6 +46,10 @@ message RepeatedFields {
   extensions 18 to 34;
 }
 
+message PropertyNamingCustomName {
+    optional string custom_name = 1 [json_name = "custom-name"];
```

Comment:
I think we got kind of lucky that the naming strategy doesn't happen to transform `"custom-name"` since it's hyphenated. Can you also add two more fields like:
```proto
optional string lower_camel = 2 [json_name = "lowerCamel"];
optional string lower_underscore = 3 [json_name = "lower_underscore"];
```

Reply:
👍 Would you expect that `json_name` would always take precedence over existing behavior?

Reply:
Yeah, I think we just want to write whatever the `json_name` is and not transform it